zerver/lib/message.py

```diff
@@ -210,14 +211,35 @@ def truncate_topic(topic_name: str) -> str:
     return truncate_content(topic_name, MAX_TOPIC_NAME_LENGTH, "...")
 
 
+def visible_edit_history_for_message(
+    message_edit_history_visibility: int,
+    edit_history: list[FormattedEditHistoryEvent],
+) -> list[FormattedEditHistoryEvent]:
+    # Makes sure that we send message edit history to clients
+    # in realms as per `message_edit_history_visibility`.
+    if (
+        message_edit_history_visibility
+        == MessageEditHistoryVisibilityEnum.MESSAGE_EDIT_HISTORY_VISIBILITY_ALL
+    ):
+        return edit_history
+
+    visible_edit_history: list[FormattedEditHistoryEvent] = []
+    for edit_history_event in edit_history:
+        if "prev_content" in edit_history_event:
+            continue
+        visible_edit_history.append(edit_history_event)
```

Comment:
Hmm, what if an edit history event contains both `prev_content` and `prev_topic`? I'm sure that's very rare, but I think it used to be possible. It would likely be correct to drop the `content/prev_content` fields from such events, rather than skipping them entirely, since the message move part should still be accessible to clients, and in fact needs to be there for the chain of `prev_topic` values to be correct.

You can likely implement some unit tests for that sort of conversion.

BTW, as a side note, https://chat.zulip.org/#narrow/channel/378-api-design/topic/edit.20history/near/2057755 would be a great project for you to pick up next.